sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/catalog/InMemoryCatalog.scala

```diff
@@ -590,6 +590,10 @@ class InMemoryCatalog(
     catalog(db).functions.remove(funcName)
   }
 
+  override protected def doAlterFunction(db: String, func: CatalogFunction): Unit = synchronized {
+    // do nothing
```

Comment:
Is there any reason not to implement this? I guess we can have `remove` and `put` here which is similar with `doRenameFunction`.